std/algorithm/searching.d

```diff
@@ -3860,6 +3861,132 @@ if (isInputRange!Range && !isInfinite!Range &&
     assert(arr.maxElement!"a.val".val == 1);
 }
 
+/** Returns an array of the minimum and maximum element in `r`.
+ * Performs `< 3n/2` comparisons, unlike the naive `< 2n`.
+ * Params:
+ *  r = The range to traverse.
+ */
+// TODO alias map = a => a
+ElementType!Range[2] extrema(Range)(Range r)
```

Comment:
The explicit `ElementType!Range` makes me wonder what happens when the elements are `const`, `immutable`, or `shared`.